src/cloud-cli/applications/deploy-app-code.ts

```diff
@@ -92,3 +102,46 @@ export async function deployAppCode(host: string): Promise<number> {
   }
 }
 
+async function buildAppInDocker(appName: string): Promise<boolean> {
+  const logger = new GlobalLogger();
+
+  // Verify Docker is running
+  try {
+    execSync(`docker > /dev/null 2>&1`)
+  } catch (e) {
+    logger.error("Docker not found.  To deploy, please start the Docker daemon and make the `docker` command runnable without sudo.")
+    return false
+  }
+
+  const dockerFileName = `${deployDirectoryName}/Dockerfile.dbos`;
+  const containerName = `dbos-builder-${appName}`;
+
+  // Dockerfile content
+  const dockerFileContent = `
+FROM node:18-alpine3.17
+RUN apk update && apk add zip
+WORKDIR /app
+COPY . .
+RUN npm clean-install
+RUN npm run build
+RUN npm prune --omit=dev
+RUN zip -ry ${appName}.zip ./* -x "${appName}.zip" -x "${deployDirectoryName}/*" > /dev/null
+`;
+  try {
+    // Write the Dockerfile
+    writeFileSync(dockerFileName, dockerFileContent);
+    // Build the Docker image.  As build takes a long time, use runCommand to stream its output to stdout.
+    await runCommand('docker', ['build', '-t', appName, '-f', dockerFileName, '.'])
```

Comment:
No more `DEPRECATED` warning about `buildx`?

Reply:
That only occurs on older versions of Docker--newer versions use `buildx` by default.